### PR TITLE
Fix UDS datagram metrics reporting

### PR DESCRIPTION
**Pull request by tobim**
* Fix the success condition for `sendto`
* Instead of shutting down on anything but timeouts we now just keep on trying to write to the socket.

This can be tested with the following `vast.yaml`:
```
vast:
  enable-metrics: true
  metrics:
    uds-sink:
      enable: true
      path: /tmp/vast-metrics.sock
      type: datagram
```
Start a vast node with this and then start and stop socat with this command:
```bash
socat  'UNIX-RECV:/tmp/vast-metrics.sock' -
```

`socat` should print data to stdout every time it is active.

### :memo: Reviewer Checklist

Review this pull request by ensuring the following items:

- [ ] All user-facing changes have changelog entries
- [ ] User-facing changes are reflected on [vast.io](https://github.com/tenzir/vast/tree/master/web)

### CODE REVIEW

**Comment by mavam** [comment]
How did you test that this fixes the issue?

File: libvast/src/system/accountant.cpp [hunk 1]
```diff
@@ -230,18 +235,16 @@ struct accountant_state_impl {
...
-          break;
-        }
-        default: {
-          VAST_WARN("{} failed to write metrics to UDS sink: {}", *self, err);
-          VAST_WARN("{} turns off metrics reporting via UDS", *self);
-          uds_datagram_sink.reset();
-          return;
-        }
+      const auto code = ec{err.code()};
+      if (code == ec::timeout)
```
Comment (dominiklohmann) [inline libvast/src/system/accountant.cpp:239]: You should be able to just say `if (err == ec::timeout)`, regardless of whether or not the error has a context in addition to the code.

File: libvast/src/detail/posix.cpp [hunk 2]
```diff
@@ -137,7 +137,7 @@ caf::error uds_datagram_sender::send(std::span<char> data, int timeout_usec) {
   // the happy path.
   if (::sendto(src_fd, data.data(), data.size(), 0,
                reinterpret_cast<sockaddr*>(&dst), sizeof(struct sockaddr_un))
-      == 0)
+      >= 0)
```
Comment (dominiklohmann) [inline libvast/src/detail/posix.cpp:140]: If the return value is the number of bytes sent, shouldn't we actually check whether it's exactly `data.size()` here and below?

**Review by dominiklohmann — APPROVED** [review]
I verified that this works by testing it locally with the provided instructions for socat.

File: libvast/src/system/accountant.cpp (hunk 1 above)
Comment (tobim) [reply libvast/src/system/accountant.cpp:239]: `ec{err.code()}` is needed a bit further down anyways so why not give it a name and avoid the indirection through an extra `operator!=`? I don't think it makes a difference in readability.

File: libvast/src/detail/posix.cpp (hunk 2 above)
Comment (tobim) [reply libvast/src/detail/posix.cpp:140]: Fair point. I'm afraid we have to come up with a new error code for that though.

File: libvast/src/system/accountant.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/system/accountant.cpp:239]: Ah, didn't see that use of it. Feel free to resolve.

**Review by lava — APPROVED** [review]
I deployed this on our testbed and it emprically fixes the UDS datagram metrics reporting.

**Comment by tobim** [comment]
[sc-37951]

**Pull request merged**